test/e2e/azure_test.go

```diff
@@ -145,6 +168,43 @@ var _ = Describe("Workload cluster creation", func() {
 				})
 			})
 		})
+
+		Context("Running the self-hosted spec", func() {
```

Comment:
why not keep this in the capi_e2e suite like it was before? It's not really testing functionality that can be regressed by CAPZ PRs

Reply:
because for the identity secret to be moved when the cluster pivots to a management cluster running on Azure, it has to be in the same namespace as the cluster and running the test spec from capi doesn't allow me to know the namespace

Reply:
right but could we define our own spec and run it as part of pull-cluster-api-provider-azure-capi-e2e ?

Reply:
oh okay, just move the calling it in capi_e2e?

Reply:
yeah that's what I meant, sorry if it wasn't clear

Reply:
no worries, that makes better sense, will change it